pkg/capi/gcp.go

```diff
@@ -0,0 +1,295 @@
+
+	return providerSpec
+}
+
+func createGCPMachineTemplate(cl client.Client, mapiProviderSpec *mapiv1.GCPMachineProviderSpec) *gcpv1.GCPMachineTemplate {
+	By("Creating GCP machine template")
+
+	Expect(mapiProviderSpec).ToNot(BeNil())
+	Expect(mapiProviderSpec.Disks).ToNot(BeNil())
+	Expect(len(mapiProviderSpec.Disks)).To(BeNumerically(">", 0))
+	Expect(mapiProviderSpec.Disks[0].Type).ToNot(BeEmpty())
+	Expect(mapiProviderSpec.MachineType).ToNot(BeEmpty())
+	Expect(mapiProviderSpec.NetworkInterfaces).ToNot(BeNil())
+	Expect(len(mapiProviderSpec.NetworkInterfaces)).To(BeNumerically(">", 0))
+	Expect(mapiProviderSpec.NetworkInterfaces[0].Subnetwork).ToNot(BeEmpty())
+	Expect(mapiProviderSpec.ServiceAccounts).ToNot(BeNil())
+	Expect(mapiProviderSpec.ServiceAccounts[0].Email).ToNot(BeEmpty())
+	Expect(mapiProviderSpec.ServiceAccounts[0].Scopes).ToNot(BeNil())
+	Expect(len(mapiProviderSpec.ServiceAccounts)).To(BeNumerically(">", 0))
+	Expect(mapiProviderSpec.Tags).ToNot(BeNil())
+	Expect(len(mapiProviderSpec.Tags)).To(BeNumerically(">", 0))
+
+	var rootDeviceType gcpv1.DiskType
+
+	switch mapiProviderSpec.Disks[0].Type {
+	case "pd-standard":
+		rootDeviceType = gcpv1.PdStandardDiskType
+	case "pd-ssd":
+		rootDeviceType = gcpv1.PdSsdDiskType
+	}
+	/*case "local-ssd":
+	rootDeviceType = gcpv1.LocalSsdDiskType*/
+	/*case "pd-balanced":
+	rootDeviceType = gcpv1.PdBalancedDiskType*/
+	//Above cases are not include Pdbalanced needs vendor file update, local-ssd not supported for boot disks
+
+	ipForwardingDisabled := gcpv1.IPForwardingDisabled
+
+	gcpMachineSpec := gcpv1.GCPMachineSpec{
+		RootDeviceType:    &rootDeviceType,
+		RootDeviceSize:    mapiProviderSpec.Disks[0].SizeGB,
+		InstanceType:      mapiProviderSpec.MachineType,
+		OnHostMaintenance: (*gcpv1.HostMaintenancePolicy)(&mapiProviderSpec.OnHostMaintenance),
+		Image:             &mapiProviderSpec.Disks[0].Image,
+		Subnet:            &mapiProviderSpec.NetworkInterfaces[0].Subnetwork,
+		ServiceAccount: &gcpv1.ServiceAccount{
+			Email:  mapiProviderSpec.ServiceAccounts[0].Email,
+			Scopes: mapiProviderSpec.ServiceAccounts[0].Scopes,
+		},
+		// Shielded VM Configuration
+		ShieldedInstanceConfig: &gcpv1.GCPShieldedInstanceConfig{
+			SecureBoot:                       gcpv1.SecureBootPolicy(mapiProviderSpec.ShieldedInstanceConfig.SecureBoot),
+			VirtualizedTrustedPlatformModule: gcpv1.VirtualizedTrustedPlatformModulePolicy(mapiProviderSpec.ShieldedInstanceConfig.VirtualizedTrustedPlatformModule),
+			IntegrityMonitoring:              gcpv1.IntegrityMonitoringPolicy(mapiProviderSpec.ShieldedInstanceConfig.IntegrityMonitoring),
+		},
+
+		// Confidential VM Configuration
+		ConfidentialCompute:   (*gcpv1.ConfidentialComputePolicy)(&mapiProviderSpec.ConfidentialCompute),
+		AdditionalNetworkTags: mapiProviderSpec.Tags,
+		AdditionalLabels:      gcpv1.Labels{fmt.Sprintf("kubernetes-io-cluster-%s", clusterName): "owned"},
+		IPForwarding:          &ipForwardingDisabled,
+	}
```

Comment:
The default machinetemplate doesn't have these parameters, right? will it affect other tests? Can we create a default machinetemplate without making any changes here, and change it after the default machinetemplate is created.

Reply:
No it wont I ran all will share here with latest commits.